This new branch was branch off Profile-Model #7 and merged with the initial branch of Profile Controller

seyeonigbinde opened the pull request:
### What this PR does (required):
- The branch was created off the Profile Model Branch and merge with the initial branch of Profile Controller. This was to correct the initial branching error
- Destructured the req.body of the `POST` and `PUT` request in issue #44 
- Validated `req.params.id` in issue #44 
- Tested all the profile endpoints  in #44  on `Postman`

### Screenshots / Videos (front-end only):
- 
<img width="1065" alt="Screenshot 2021-10-20 at 07 03 29" src="https://user-images.githubusercontent.com/79403650/138055114-b73c3eb0-fb4a-4426-8d94-fcdb406c5e58.png">

rajivtitus commented on server/controllers/profile.js:
```diff
@@ -0,0 +1,57 @@
+exports.getProfileById = asyncHandler(async (req, res, next) => {
+  const profileId = req.params.id 
+  await Profile.findById(profileId, (profile) => {
+    if (!mongoose.Types.ObjectId.isValid(profileId)) {
```
This check needs to be performed after line 16 , not within this callback

rajivtitus commented on server/controllers/profile.js:
```diff
@@ -0,0 +1,57 @@
+      description },
+    {new: true},
+    (profile) => {
+    if (!mongoose.Types.ObjectId.isValid(profileId)) {
+    return res.status(400).send("Bad Request");
```
Same as comment above!

seyeonigbinde commented:
- Validated the `req.body `of `POST` and `PATCH` in #6  
- Changed `PUT` method to `PATCH` in #6  
- Adjusted the positioning of `req.params.id` validation in #6  
- @rajivtitus Kindly review.
